### ISSUE TITLE
TFsec doesn't recognize terraform v1.1.0 modules after init

### ISSUE THREAD

**Issue by wilkej** [aquasecurity/tfsec#1188]
**Describe the bug**
We've got terraform and tfsec running in a gitlab pipeline. Recently I updated to TF 1.1.0 image and now TF sec shows the error

```hcl 
WARNING: Failed to load module: missing module with source 'terraform-aws-modules/s3-bucket/aws' -  try to 'terraform init' first
WARNING: Failed to load module: missing module with source 'terraform-aws-modules/eks/aws' -  try to 'terraform init' first
```
If I change the TF image in the pipeline to 1.0.11 it works. 

**To Reproduce**

1. run terraform 1.1.0 init
2. run tfsec and you'll see the errors

1. delete .terraform folder
2. run terraform 1.0.11 init
3. run tfsec you'll not see the errors

**Expected behavior**
TFsec should find the init modules for TF 1.1.0 

**Desktop (please complete the following information):**
 - OS: Ubuntu Linux

**Comment by fgm** [comment]
Error also happens with default modules like `hashicorp/subnets`

**Comment by martijnvdp** [comment]
looks like the source changed from 1.1.0 > example:
before : `{"Key":"endpoint_sg","Source":"**terraform-aws-modules/security-group/aws**","Version":"4.2.0","Dir":".terraform/modules/endpoint_sg"}`
now
`{"Key":"endpoint_sg","Source":"**registry.terraform.io/terraform-aws-modules/security-group/aws**","Version":"4.2.0","Dir":".terraform/modules/endpoint_sg"}`

**Comment by martijnvdp** [comment]
hashicorps resolves to registry.terraform.com see  also : https://github.com/hashicorp/terraform/blob/main/internal/addrs/module_source_test.go
ive made a pr to change the check from source to module key name , but probably there other options/solutions 
also looked at go-getter but that uses a lot of dependencys

**Comment by sergei-ivanov** [comment]
It is probably related to this major change in TF 1.1.x: https://github.com/hashicorp/terraform/pull/28854

**Comment by sergei-ivanov** [comment]
@liamg  Can this be closed, now that #1206 has been merged?
And together with the fix for #1192, can we say that the latest RC versions are fully compatible with TF 1.1?
On my side, I put TF 1.1 upgrade on hold, waiting for TFSec to add first class support. Do you need help testing the RC, or is it not ready for early testing yet?

**Comment by liamg** [comment]
@sergei-ivanov I believe so, yes, on both counts :+1: 

Help testing the RC would be very welcome and much appreciated! We're looking to get as much real-world testing as possible before tagging v1, but we have a lot of unreleased code so don't want to delay too long either.
